pkg/deploy/controller/deploymentconfig/controller.go

```diff
@@ -382,3 +404,59 @@ func (c *DeploymentConfigController) handleErr(err error, key interface{}) {
 		c.queue.Forget(key)
 	}
 }
+
+// cleanupOldDeployments deletes old replication controller deployments if their quota has been reached
+func (c *DeploymentConfigController) cleanupOldDeployments(existingDeployments []kapi.ReplicationController, deploymentConfig *deployapi.DeploymentConfig) error {
+	if deploymentConfig.Spec.RevisionHistoryLimit == nil {
+		// there is no past deplyoment quota set
+		return nil
+	}
+
+	activeDeployment := deployutil.ActiveDeployment(deploymentConfig, existingDeployments)
+
+	// we need to coerce the list of deployments we want to consider into something we can sort
+	oldDeployments := []*kapi.ReplicationController{}
+	if activeDeployment == nil {
+		// if cleanup policy is set but no successful deployments have happened, there will be
+		// no active deployment. We can consider all of the deployments in this case
```

Comment:
We should never consider the latest deployment or the active deployment (latest+plus rolledback deployment) for cleaning up